### PR TITLE
Add script to auto-upload all reports to the right google sheet

### PR DESCRIPTION
**Pull request by gruberb**
Added a script to auto-upload the generated `reports` to the right Google Sheets and overwrites them.

This PR adds a 
```bash
make run_google_sheet_uploader
```

command to automatically go through all the generated reports and uploads them to the respective Google Sheets (and tabs).

Closes: #[ECTEEN-125](https://mozilla-hub.atlassian.net/browse/ECTEEN-125)

[ECTEEN-125]: https://mozilla-hub.atlassian.net/browse/ECTEEN-125?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: docs/metric-update-procedure/README.md [hunk 1]
```diff
@@ -54,23 +88,16 @@ _**Notes**:_
...
-- In the top menu bar, select `File > Import`, a file selection dialogue should appear
-- Select the `Upload` tab and click `Browse`, a local file selection dialogue should appear
-- Select the corresponding csv file from the `reports_dir`, an `Import file` dialogue should
-  appear
-- For `Import location`, select `Replace current sheet`, for `Separator type,` select
-  `Detect automatically`, leave the `Convert text to numbers, dates, and formulas` box
-   checked, and click `Import data.` The Google Sheet should now reflect the latest data.
 - If the report being imported is a results or coverage report. Convert the type of the date
-     column to 'plain text' so that the graphs display at an even cadence. 
+     column to 'plain text' so that the graphs display at an even cadence.
```
Comment (gruberb) [inline docs/metric-update-procedure/README.md:100]: Does this still need to happen? As well as `3.3 Update Trend Table Dates`? If so, we can probably add this to the script as well.

File: config.ini.sample [hunk 2]
```diff
@@ -60,3 +60,163 @@ days_of_data = 2
...
+        },
+        {
+            "file_name": "autopushrs_integration_tests_results.csv",
+            "sheet_id": "1abjtg2e-PHm8JDP5A629KFVA-eBc8I5VtFPcz_UCDQs",
+            "tab_name": "Integration Results"
+        },
+        {
+        "file_name": "autopushrs_rust_unit_tests_averages.csv",
+            "sheet_id": "1abjtg2e-PHm8JDP5A629KFVA-eBc8I5VtFPcz_UCDQs",
+            "tab_name": "Unit Averages"
```
Comment (Trinaa) [inline config.ini.sample:80]:
```suggestion
            "file_name": "autopushrs_rust_unit_tests_averages.csv",
            "sheet_id": "1abjtg2e-PHm8JDP5A629KFVA-eBc8I5VtFPcz_UCDQs",
            "tab_name": "Unit Averages"
```

File: config.ini.sample [hunk 3]
```diff
@@ -60,3 +60,163 @@ days_of_data = 2
 
 [metric_reporter]
 reports_dir = reports
+
+[google_sheet_uploader]
+service_account_file = google_sheets_key.json
+mappings = [
```
Comment (Trinaa) [inline config.ini.sample:66]: I wonder if there is a structure we could use so we don't need to repeat the sheet_id?

File: config.ini.sample [hunk 4]
```diff
@@ -60,3 +60,163 @@ days_of_data = 2
...
+        },
+        {
+            "file_name": "fxa_firefox_functional_tests_playwright_nightly_averages.csv",
+            "sheet_id": "1qKwxsSI2RNo-qKZflETtBbyZ9b0ZUB_id2SL17_MmYo",
+            "tab_name": "Nightly Firefox Functional Averages"
+        },
+        {
+            "file_name": "fxa_firefox_functional_tests_playwright_nightly_results.csv",
+            "sheet_id": "1qKwxsSI2RNo-qKZflETtBbyZ9b0ZUB_id2SL17_MmYo",
+            "tab_name": "Nightly Integration Frontends Averages"
```
Comment (Trinaa) [inline config.ini.sample:105]:
```suggestion
            "tab_name": "Nightly Firefox Functional Results"
```

File: config.ini.sample [hunk 5]
```diff
@@ -60,3 +60,163 @@ days_of_data = 2
...
+        },
+        {
+            "file_name": "fxa_integration_test_frontends_nightly_averages.csv",
+            "sheet_id": "1qKwxsSI2RNo-qKZflETtBbyZ9b0ZUB_id2SL17_MmYo",
+            "tab_name": "Nightly Integration Frontends Results"
```
Comment (Trinaa) [inline config.ini.sample:110]:
```suggestion
            "tab_name": "Nightly Integration Frontends Averages"
```

**Review by Trinaa — COMMENTED** [review]
@gruberb I tried this out and it's super awesome. 

I think we need to change the line `The process of updating all three Google Sheets can take up to 30 minutes.` to something like 10 minutes now 🚀 

Reminder - Make sure to format your commit messages according to the guidelines with the `Closes` string, this makes sure our JIRA integration works correctly.

File: docs/metric-update-procedure/README.md (hunk 1 above)
Comment (gruberb) [reply docs/metric-update-procedure/README.md:100]: I think we can add this once the basic uploading works as expected and is merged into `main`.

File: docs/metric-update-procedure/README.md [hunk 6]
```diff
@@ -8,15 +8,49 @@ Please follow the steps below to update the following Google Sheets:
...
-to 30 minutes.
+to 10 minutes.
+
+## First set up (just has to be done once)
+
+### Prepare the auto-uploader to Google Sheets
+
+#### Step 1: Create a Google Cloud Project
+* Go to the [Google Cloud Console](https://console.cloud.google.com/apis/dashboard?project=ecosystem-test-eng).
+* Create a new project or select an existing one.
```
Comment (Trinaa) [inline docs/metric-update-procedure/README.md:19]: nit:

```suggestion
#### Step 1:Navigate to the ETE Google Cloud Project
* Go to the [Google Cloud Console](https://console.cloud.google.com/apis/dashboard?project=ecosystem-test-eng).
```

File: scripts/google_sheet_uploader/uploader.py [hunk 7]
```diff
@@ -0,0 +1,156 @@
...
+        self.config = config
+        self.logger = logging.getLogger(__name__)
+
+        try:
+            self.credentials = service_account.Credentials.from_service_account_file(
+                self.config.service_account_file,
+                scopes=["https://www.googleapis.com/auth/spreadsheets"],
+            )  # type: ignore
+            self.service = build("sheets", "v4", credentials=self.credentials)
+        except Exception as e:
```
Comment (Trinaa) [inline scripts/google_sheet_uploader/uploader.py:59]: nit: Do you happen to know which errors exactly we should catch? Catchalls aren't generally a good idea.

**Review by Trinaa — APPROVED** [review]
LGTM! Can't wait to use this on Monday.

I made two minor nits that I don't consider to be blockers, just wanted to share my thoughts.

**Pull request merged**